#!/usr/bin/env python
PACKAGE = "stroll_bearnav"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thresholdParam", int_t, 0, "A threshold for SURF hessian detector", 400, 0,  100000)
#gen.add("ratioMatchConstant", double_t, 0, "Constant for ratio matching", 0.7, 0,  1)
gen.add("adaptThreshold",       bool_t,    0,    "Adaptive threshold on", True)
gen.add("targetKeypoints",       int_t,    0,    "Desired number of keypoints", 60,     0, 10000)
gen.add("gain",       double_t,    0,    "Target gain", 0.3,     0, 3)
gen.add("showImage",       bool_t,    0,    "Show Images with features", True)
gen.add("optimized",       bool_t,    0,    "First detect, then compute", False)
gen.add("measure_time",       bool_t,    0,    "Measuring time on", False)
exit(gen.generate(PACKAGE, "stroll_bearnav", "featureExtraction"))
